### ISSUE TITLE
List of aperture_photometry's additional future features. 

### ISSUE THREAD

**Issue by bsipocz** [astropy/photutils#131]
These are optional todos/new features for aperture_photometry. I'll update the list based on the comments, etc.:
- [x] add rectangular_annulus, see #149
- [ ] add polygon aperture class and do_polygon_photometry function
- [x] accept non pixel unit for radius etc. when the wcs is provided, see #138

**Comment by bsipocz** [comment]
Actually there is a demand for polygon apertures, so this can be a feature to work on if someone has unused resources. Making the base class should be quite straightforward, but providing a ``method='exact'`` version may need some extra hustle.

**Comment by cheekujodhpur** [comment]
Hi! I am trying to join the AstroPy developer community. I would like to start with this.

Do you think you can assign this to me and help me with more information? I am interested.

**Comment by cheekujodhpur** [comment]
Hi! I am interested to work on this issue. I am one of the people who demand polygon apertures, and also I would like to start with my contribution to the astropy community.

Do you think you can assign me this?

**Comment by hamogu** [comment]
Leaving your comment here is enough. No need to assign in addition to that.

**Comment by cheekujodhpur** [comment]
I was unable to use `do_photometry`, and we lack an example in apertures.rst
My `data` for test purposes is `np.zeros(100,100)`
but the `do_circular_photometry` code has this line of code

`flux = u.Quantity(np.zeros(len(positions), dtype=np.float), unit=data.unit)`
Now, naturally `data` has no attribute called `unit`. What kind of object do you expect data to be here?

**Comment by cheekujodhpur** [comment]
Nevermind a few comments before. I had some queries which I figured out by myself now.
Before I start working on the PolygonAperture, I wanted to discuss if it is a good idea to define TriangleAperture first and treat a polygon as sum of triangles?

**Comment by larrybradley** [comment]
@bsipocz I'm going to close this an create an new issue specifically about adding a polygon aperture (once `regions` is ready).  Thanks.
